Game2048Base/src/main/java/models/grid2048/Grid2048Event.java

```diff
@@ -0,0 +1,27 @@
+package models.grid2048;
+
+import models.game2048.KeyEventHandler;
+import models.game2048.Scoreboard;
+
+public enum Grid2048Event {
+    UP, DOWN, LEFT, RIGHT;
+
+    public KeyEventHandler buildKeyEventHandlerOnKeyEvent(Grid2048 grid, Scoreboard scoreboard) {
```

Comment:
This is an atypical way to use an enum data type and probably doesn't belong in here.

The enum is a way for you to restrict a data set.

This class should only really be UP, DOWN, LEFT, RIGHT, and maybe it will be helpful to add a keyevent string to the constructor to make the comparison easier because it's defined here. The handler methods should take a keyevent as input, but should not be a instance _of_ a keyevent